### PR TITLE
fix: Api new line chars getting lost issue fixed

### LINKED ISSUE #18709: [Bug]-[35840]:API Headers - new line character is lost after Navigating to another page and returning to the API
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I reference an Input (or any widget) in an API, and the reference has new line character in it, if I move away to another page and get back to the API, it is seen that the new line is lost. 

### Steps To Reproduce

1. Create a new Page, say Page3, and drag and drop an input widget say Input1
2. Create a new api (Api1) and in the Headers tab, in the Value section, add the binding 
{{
Input1.text
}}
3. Navigate to Page1, and then return to Api1. 
4. Note that the binding is now showing up as {{Input1.text}}
The new line characters are lost

### Public Sample App

_No response_

### Issue video log

https://ddpa7mfuvv.vmaker.com/record/m0dAXHeTAitF7CWk

### Version

Production

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR fixes:
- When creating REST API query, if in the headers, we add text with newline characters, and now if we switch to another page and come back to this page again, we can see the new line characters have been lost from API headers.
- This issue was also there in case API query parameters and when setting API body with FORM_URLENCODED and MULTIPART_FORM_DATA types, fixed here as well.

> Add a TL;DR when description is extra long (helps content team)

Fixes #18709 , #18744
> if no issue exists, please create an issue and ask the maintainers about this first

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Jest

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/components/editorComponents/CodeEditor/index.tsx [hunk 1]
```diff
@@ -155,6 +155,7 @@ export type EditorStyleProps = {
   popperPlacement?: Placement;
   popperZIndex?: Indices;
   blockCompletions?: FieldEntityInformation["blockCompletions"];
+  preserveNewLineChars?: boolean;
```
Comment (hetunandu) [inline app/client/src/components/editorComponents/CodeEditor/index.tsx:158]: Can we create a new "size" type that will handle this and not make it a new prop?

Comment (sneha122) [reply app/client/src/components/editorComponents/CodeEditor/index.tsx:158]: Yes created new editor size type to preserve formatting, Please check

**Review by hetunandu — DISMISSED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**